src/main/java/net/imagej/ops/descriptor3d/AbstractPolygon.java

```diff
@@ -0,0 +1,136 @@
+package net.imagej.ops.descriptor3d;
+
+import java.util.List;
+
+/**
+ * An {@link AbstractPolygon} consists of vertices and neighbors.
+ * 
+ * @author Tim-Oliver Buchholz, University of Konstanz.
+ *
+ */
+public abstract class AbstractPolygon {
```

Comment:
My opinionated suggestions about class naming:

* If you want to use a class as the entry point into a data structure, don't prefix `Abstract`. Needing to write e.g. `List<AbstractFoo>` is ugly and unnecessary. Just name it `Foo` and write `List<Foo>` in that case.
* It is bad design to name a class `AbstractFoo` unless there is a `Foo` interface it implements. And in that case, you should type things on `Foo` and never `AbstractFoo`. The `AbstractFoo` class is an _SPI_-level class, intended for downstream _implementors_, but not downstream API _consumers_.